src/dlgConnectionProfiles.cpp

```diff
@@ -1875,26 +1875,25 @@ bool dlgConnectionProfiles::validateProfile()
             valid = false;
         }
 
-        if (url.indexOf(QRegularExpression(qsl("^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$")), 0) != -1) {
-            if (port_ssl_tsl->isChecked()) {
-                notificationAreaIconLabelError->show();
-                notificationAreaMessageBox->setText(
-                        qsl("%1\n%2\n\n%3").arg(notificationAreaMessageBox->text(), tr("Please enter the URL or IP address of the Game server."), check.errorString()));
-                host_name_entry->setPalette(mErrorPalette);
-                validUrl = false;
-                valid = false;
-            }
-        }
-
-        if (url.indexOf(QRegularExpression(qsl("^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$")), 0) != -1) {
-            if (port_ssl_tsl->isChecked()) {
-                notificationAreaIconLabelError->show();
-                notificationAreaMessageBox->setText(
-                        qsl("%1\n%2\n\n%3").arg(notificationAreaMessageBox->text(), tr("SSL connections require the URL of the Game server."), check.errorString()));
-                host_name_entry->setPalette(mErrorPalette);
-                validUrl = false;
-                valid = false;
-            }
+        // Need to reject raw IP addresses (of either version 4 or 6 type) as
+        // it is very unlikely that the Security Certificates include them as
+        // a Host Name.
+        if (port_ssl_tsl->isChecked() && (cTelnet::isRawIPv4Address(url) || cTelnet::isRawIPv6Address(url))) {
+            notificationAreaIconLabelError->show();
+            // As the only tags are not on the first line the default
+            // Qt::AutoFormat won't detect that rich-text is present in this text!
+            notificationAreaMessageBox->setTextFormat(Qt::RichText);
+            /*: Please use two line-feeds after the first line so the second
+             *  line can be italicised and spaced out - if appropriate for
+             *  the locale.
+             */
+            notificationAreaMessageBox->setText(qsl("%1%2\n\n%3").arg(!notificationAreaMessageBox->text().isEmpty() ? notificationAreaMessageBox->text().append(QChar::LineFeed) : QString(),
+                                                                      tr("Please enter the URL of the Game server.\n\n"
+                                                                         "<i>SSL connections require the URL of the Game server.</i>"),
```

Comment:
This changed text now seems pretty redundant.

Also SSL is deprecated, TLS the new way to go.
Anyway, too much technical detail for players.

```suggestion
                                                                      tr("For secure connections, please enter the URL of the Game server."),
```

Reply:
The "certificate" used for the encryption is extremely unlikely to include the IP Address as an identifier for the Server - indeed we did reject any "raw" (quad dot separated numbers each between 0 and 255) IPv4 addresses if that was previously supplied as the "URL" for a "secure" connection in the past (this PR extends that to also reject raw IPv6 ones in the same way) - anyhow this is the warning message for the user so this explanation (without the technical reasons behind it) why Mudlet is rejecting what was entered seems reasonable IMHO. 

I'd consider changing `SSL` to `SSL/TLS` but I think it sounds/looks better to say that rather than reusing "secure" yet again - besides even though SSL is deprecated that doesn't stop it from being used!